src/lib.rs

```diff
@@ -2,6 +2,10 @@
 #![warn(missing_docs)]
 #![warn(rustdoc::missing_crate_level_docs)]
 #![warn(unreachable_pub)]
+#![warn(missing_debug_implementations)]
+#![warn(clippy::pedantic)]
+#![warn(clippy::cargo)]
+#![cfg_attr(test, deny(warnings))]
```

Comment:
Let's remove line `#![cfg_attr(test, deny(warnings))]` 8 because I want people to be able to test even if they have warnings and CI will deny warnings anyway (but not for tests...)